src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java

```diff
@@ -302,12 +302,17 @@ private IMessage convertWithoutDictionary(Map<String, Value> fieldsMap, String m
     }
 
     private Object traverseField(String fieldName, Value fieldValue) {
-        if (fieldValue.hasMessageValue()) {
-            return convertWithoutDictionary(fieldValue.getMessageValue().getFieldsMap(), fieldName);
-        } else if (fieldValue.hasListValue()) {
-            return convertList(fieldName, fieldValue.getListValue());
+        switch (fieldValue.getKindCase()) {
+            case SIMPLE_VALUE:
+                return fieldValue.getSimpleValue();
+            case MESSAGE_VALUE:
+                return convertWithoutDictionary(fieldValue.getMessageValue().getFieldsMap(), fieldName);
+            case LIST_VALUE:
+                return convertList(fieldName, fieldValue.getListValue());
+            case NULL_VALUE:
+            default:
+                return null;
```

Comment:
Please, throw an exception in the default case instead. Otherwise, we might bury the problems